### ISSUE TITLE
When mocking a function that has a JavaScript syntax error, nothing happens, endless wait

### ISSUE THREAD

**Issue by berenddeboer** [aws-amplify/amplify-cli#5159]
**Describe the bug**
Make a syntax error in your `index.js`. Then mock the function. Mock will hang until you press Ctrl+C.

Then you get the useless:

```
Starting execution...
^Cundefined:1



SyntaxError: Unexpected end of JSON input
at JSON.parse (<anonymous>)
at ChildProcess.<anonymous> (/usr/lib/node_modules/@aws-amplify/cli/node_modules/amplify-nodejs-function-runtime-provider/lib/utils/invoke.js:45:37)
at ChildProcess.emit (events.js:315:20)
at ChildProcess.EventEmitter.emit (domain.js:483:12)
at maybeClose (internal/child_process.js:1021:16)
at Socket.<anonymous> (internal/child_process.js:443:11)
at Socket.emit (events.js:315:20)
at Socket.EventEmitter.emit (domain.js:483:12)
    at Pipe.<anonymous> (net.js:674:12)
```

**Amplify CLI Version**
4.27.3

**To Reproduce**
Make a syntax error in your `index.js`. Then mock.

**Expected behavior**
The actual error returned by nodejs.

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04
 - Node Version: 12.18.3

**Comment by berenddeboer** [comment]
Might be fixed by #4978.

**Comment by devyzr** [comment]
> Might be fixed by #4978.

It doesn't for me

**Comment by dmsolutionz** [comment]
Version 4.24.3 was the last known good version where even simple logging events worked properly, now mocking will force a hang, or give unusable logging / error messaging. Basically making our amplify project currently locked to 4.24.3...

**Comment by kaustavghosh06** [comment]
We've fixed this issue as a part of #5198
